src/import-data/useImportRequest.ts

```diff
@@ -0,0 +1,136 @@
+import { useRoute } from 'src/libs/nav';
+
+import {
+  CatalogDatasetImportRequest,
+  CatalogSnapshotsImportRequest,
+  FileImportRequest,
+  ImportRequest,
+  TDRSnapshotExportImportRequest,
+  TDRSnapshotReferenceImportRequest,
+} from './import-types';
+
+type QueryParams = { [key: string]: unknown };
+
+/**
+ * Validate that a value is a string. Throw an error if it is not.
+ *
+ * @param value The value to validate.
+ * @param label Label for the value to use in error messages.
+ * @returns The validated string value.
+ */
+const requireString = (value: unknown, label = 'value'): string => {
+  if (value === undefined || value === null) {
+    throw new Error(`A ${label} is required`);
```

Comment:
I'm new to javascript exceptions, is there no better error type for this case, e.g., `IllegalArgument`?

Reply:
JS doesn't have many built-in error types.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#error_types

We could define a custom error type.

Reply:
actually, this is an area where terra-ui codebase is weak....  AoU makes use of a more robust custom Error pattern that we should totally look at adopting.